pkgs/development/python-modules/stestr/default.nix

```diff
@@ -0,0 +1,42 @@
+{ lib, buildPythonPackage, fetchPypi, callPackage
+, future, pbr, cliff, subunit, fixtures, testtools, pyyaml, voluptuous
+}:
+
+buildPythonPackage rec {
+  pname = "stestr";
+  version = "3.2.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "fb492cbdf3d3fdd6812645804efc84a99a68bb60dd7705f15c1a2949c8172bc4";
+  };
+
+  propagatedBuildInputs = [
+    future
+    pbr
+    cliff
+    subunit
+    fixtures
+    testtools
+    pyyaml
+    voluptuous
+  ];
+
+  # Tests should only run on passthru to avoid infinite recursion from `openstack-hacking`.
```

Comment:
Other ways people sometimes achieve this is to use an overridden version of one of the packages with `doCheck` disabled _just for use in the tests_. The main version seen by the end user of course does have the tests enabled.